README.md

```diff
@@ -4,13 +4,57 @@ A new Flutter project.
+variables. The default values could be used for local development purposes, and CI/CD tools can override the values 
+depending on the environment being deployed to.
 
-A few resources to get you started if this is your first Flutter project:
+Simply run:
 
-- [Lab: Write your first Flutter app](https://flutter.dev/docs/get-started/codelab)
-- [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)
+```sh
+flutter run  # to rely on the default 'multiplier' value, or
+flutter run --dart-define multiplier=2  # to specify the 'multiplier' value
+```
 
-For help getting started with Flutter, view our
-[online documentation](https://flutter.dev/docs), which offers tutorials,
-samples, guidance on mobile development, and a full API reference.
+More about `--dart-define` can be found [here](https://itnext.io/flutter-1-17-no-more-flavors-no-more-ios-schemas-command-argument-that-solves-everything-8b145ed4285d).
+
+## How might we use this?
+
+While some configuration will be dynamic during the lifetime of the app, in our case likely resolved from Firebase
+Remote Config, other configuration is static and constant, and can be decided at the time of build / deployment. For
+this kind of configuration, we can easily leverage `--dart-define` arguments and `String.fromEnvironment` calls to
+define what the 'build-time' configuration is. This will be particularly powerful if, in our app's configuration 
+library, we can just expose the latest config to client code, and hide the details of where exactly the config is
+sourced from. For example, to borrow some concepts from `wooliesgo_flutter`:
+
+```dart
+class MyClass {
+    final AlwaysCurrentConfig _config;
+
+    MyClass({ required AlwaysCurrentConfig config }) : _config = config;
+
+    void doSomething() {
+        if (_config.latest.someFeatureIsEnabled) {
+            _sendSomethingTo(_config.latest.someApiUrl)
+        }
+    }
+}
+```
+
+The provider of the `someFeatureIsEnabled` property would be dynamic, and so calling `_myClass.doSomething()` would
+have different results depending on whether the feature has been enabled for the current user, but anyone using the
+app in the same environment would get the same value for `someApiUrl`.
+
+Then it's a decision of how granular we want build- / deploy-time configuration to be. There's...
+
+1. **Broad stroke:** We could just specify the intended environment, i.e. `dev`, `uat` or `prod` and our code could 
+    know which urls / values that should map to. This has the advantage of not requiring changes to our CI/CD script 
+    each time we have a new configurable property, but the downside of the code needing to know about configuration it 
+    shouldn't use in a given environment - it also cannot handle the case where we may need to keep keys / secrets out 
+    of our source control, so we'd still need to provide sensitive data like this as `--dart-define` arguments.
```

Comment:
I know that `someApiUrl` is just an example but currently we dynamically get these urls from Apigee at `getRouteDetails` which got me thinking that maybe we need to list what specific configuration we would use this for and that may help us with the decision of walking this path or not

Reply:
The Apigee url itself, though, is known at build time (in the Android `app/build.gradle`, we have a `buildConfigField` that defines `"BASE_URL"`), so that could be the `someApiUrl` that we have in this example.

> maybe we need to list what specific configuration we would use this for and that may help us with the decision of walking this path or not

I'm not entirely sure what you mean by this. If it's a case of "what kind of configuration would we use `--dart-define` for", then I'd summarise the answer as "anything we want configurable where the value is known at build time". Instead of planning to maintain some fairly static configuration in Remote Config (config that isn't specific to an audience and shouldn't change for a given version of the app), we can instead go with the simpler `--dart-define` solution for that particular config property.